tests/test_core/test_to_dict.py

```diff
@@ -0,0 +1,90 @@
+import pickle
+from json import JSONDecoder
+
+import hydra_slayer
+import pytest
+from ruptures import Binseg
+from sklearn.linear_model import LinearRegression
+
+from etna.core import BaseMixin
+from etna.ensembles import StackingEnsemble
+from etna.ensembles import VotingEnsemble
+from etna.metrics import MAE
+from etna.models import CatBoostModelPerSegment
+from etna.models import LinearPerSegmentModel
+from etna.pipeline import Pipeline
+from etna.transforms import AddConstTransform
+from etna.transforms import ChangePointsTrendTransform
+from etna.transforms import LogTransform
+
+
+@pytest.fixture()
+def ensemble_samples():
+    pipeline1 = Pipeline(
+        model=CatBoostModelPerSegment(),
+        transforms=[
+            AddConstTransform(in_column="target", value=10),
+            ChangePointsTrendTransform(
+                in_column="target", change_point_model=Binseg(), detrend_model=LinearRegression(), n_bkps=50
+            ),
+        ],
+        horizon=5,
+    )
+    pipeline2 = Pipeline(
+        model=LinearPerSegmentModel(),
+        transforms=[
+            ChangePointsTrendTransform(
+                in_column="target", change_point_model=Binseg(), detrend_model=LinearRegression(), n_bkps=50
+            ),
+            LogTransform(in_column="target"),
+        ],
+        horizon=5,
+    )
+    ensemble1 = VotingEnsemble(pipelines=[pipeline1, pipeline2], weights=[0.4, 0.6])
+    ensemble2 = StackingEnsemble(pipelines=[pipeline1, pipeline2])
+    return ensemble1, ensemble2
+
+
+class _InvalidParsing(BaseMixin):
+    def __init__(self, a: JSONDecoder):
+        self.a = a
+
+
+@pytest.mark.parametrize(
```

Comment:
Can we separate test cases for models, metrics, transforms only in case we would have errors in the future.


We can add models like AutoArima, nn.models, some metrics, fourier transform and etc